test/date/test_date.rb

```diff
@@ -175,4 +175,23 @@ def test_infinity_comparison
     assert_equal(-1, -Float::INFINITY <=> Date::Infinity.new)
     assert_equal(-1, -Date::Infinity.new <=> Float::INFINITY)
   end
+
+  def test_deconstruct_keys
+    d = Date.new(1999,5,23)
+    assert_equal({year: 1999, month: 5, day: 23, wday: 0, yday: 143}, d.deconstruct_keys(nil))
+    assert_equal({year: 1999}, d.deconstruct_keys([:year, :century]))
```

Comment:
Can you add an assertion that tests for all valid keys, both for Date and DateTime, so all branches get covered?